Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraintPalmUp.cs

```diff
@@ -163,5 +197,125 @@ protected override bool IsValidController(IMixedRealityController controller)
 
             return true;
         }
+
+        /// <summary>
+        /// Checks to see if the palm is currently facing the user
+        /// </summary>
+        /// <param name="jointedHand"></param>
+        /// <param name="palmPose"></param>
+        /// <param name="palmCameraAngle"></param>
+        /// <returns></returns>
+        private bool IsPalmFacingUpwards(IMixedRealityHand jointedHand, MixedRealityPose palmPose, float palmCameraAngle)
```

Comment:
The name of this function isn't quite right. You're checking if the Palm Facing Constraints are met.
There's two things happening here - FlatHand check, and palm facing camera.

Reply:
That's fair; I was conflicted on what to name that for those exact reasons. I ended up with this because it was consistent with the name of the class (HandConstraintPalmUp)